### PR TITLE
feat: Switch from Namefilter to Name Accumulators

### PR DESCRIPTION
**Pull request by matheus23**
Work on name accumulators (spec WIP: https://github.com/wnfs-wg/spec/pull/65).

### CODE REVIEW

**Comment by matheus23** [comment]
Final TODOs:
- [ ] Document public API
- [ ] Split `wasm-nameaccumulator/src/lib.rs` into separate files
- [ ] Move single test in `forest/proofs.rs` into an example instead
- [ ] More tests in `forest/proofs.rs`

**Comment by appcypher** [comment]
This is awesome work! This should 🤞🏾 round up work on format stabilisation.
The accumulator implementation looks good to me so I don't have much comment there. Just bits on nits here and there.

Now that we have `*Forest::load/store` helper functions, some lines like the one below need update: 
https://github.com/wnfs-wg/rs-wnfs/blob/122fa2a0a7eb413cd66a21cae74569688e55d582/wnfs/examples/private.rs#L68

Update to `CryptoRngCore`:
https://github.com/wnfs-wg/rs-wnfs/blob/122fa2a0a7eb413cd66a21cae74569688e55d582/wnfs/src/lib.rs#L19

File: wnfs/src/private/forest/hamt.rs [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+        let name_hash = &Sha3_256::hash(&name.as_accumulator(&self.accumulator).as_ref());
+        self.get_encrypted_by_hash(name_hash, store).await
+    }
+
+    async fn remove_encrypted(
+        &mut self,
+        name: &Name,
+        store: &impl BlockStore,
+    ) -> Result<Option<Pair<NameAccumulator, BTreeSet<Cid>>>> {
+        let name_hash = &Sha3_256::hash(&name.as_accumulator(&self.accumulator).as_ref());
```
[the remark this answers is not in the document]
Comment (matheus23) [reply wnfs/src/private/forest/hamt.rs:178]: Fixed it (and another place where I did this, too)

File: wnfs/src/private/node/header.rs [hunk 2]
```diff
@@ -56,34 +51,27 @@ pub struct PrivateNodeHeader {
...
-        parent_bare_name: Namefilter,
+        parent_name: &Name,
         ratchet_seed: HashOutput,
-        inumber: HashOutput,
+        inumber: NameSegment,
     ) -> Self {
+        // A keyed hash to use for determining the seed increments without
+        // leaking info about the seed itself (from the ratchet state).
+        let seed_hash = Sha3_256::new()
+            .chain_update("WNFS ratchet increments")
```
[the remark this answers is not in the document]
Comment (matheus23) [reply wnfs/src/private/node/header.rs:69]: Ah, this one's on the chopping block anyways.

Should I just remove it in this PR?

Yeah I think I'll just remove it.
It's not useful anymore. You'd need to derive the `AccumulatorSetup` from a seed as well, and there's no such API at the moment (and I'm not planning on adding one).

**Comment by matheus23** [comment]
@appcypher I decided to remove the hash algorithm generic type from the `HamtForest`. However, this means that we can no longer do tests with a `MockHasher`.
But I decided to go ahead and remove that test. It's quite a long test (hard to maintain), and I'd expect most of the code it covers would also be covered by the proptests in wnfs-hamt.

**Review by appcypher — APPROVED** [review]
<div align="center">
    <a href="#" target="_blank">
        </img>
    </a>
</div>

<h3 align="center">LGTM! 🎉 </h3>

**Pull request merged**